src/Nelmio/Alice/Instances/Populator/Methods/ArrayAdd.php

```diff
@@ -84,4 +85,37 @@ private function findAdderMethod($object, $property)
             return $method;
         }
     }
+
+    /**
+     * @param string $plural
+     *
+     * @return string
+     */
+    private function singularize($plural)
+    {
+        if (null === $this->singularizeClass || null === $this->singularizeMethod) {
+            // default to false so we only try this once
+            $this->singularizeClass = false;
```

Comment:
Ah fair I didn't think of that use case. That said, we are using only 1 instance of `ArrayAdd`, so if left as it is you could end up using the reflection class of another class than the one checked.